doc/policies.md

```diff
@@ -0,0 +1,180 @@
+  specified that policy A comes before policy B.
+- body_filter: do nothing. Neither policy A nor B describe what to do.
+- post_action: do nothing. Neither policy A nor B describe what to do.
+- log: do nothing. Neither policy A nor B describe what to do.
+
+Notice that we did not indicate what APIcast does in the `init` and the
+`init_worker` phases. The reason is that those two are not executed in every
+request. `init` is executed when APIcast boots, and `init_worker` when each
+of each of its workers start.
+
+### Types
+
+There are two types of policy chains in APIcast: per-service chains and a
+global chain. Both of them can be configured.
+
+As the name indicates, per-service chains allow us to define a specific chain
+for a given service. This means that we can apply different policies to our
+services, or the same policies but configured differently, or in a different
+order. On the other hand, there's only one global chain, and the behavior it
+defines applies to all the services.
+
+
+## APIcast default policies
+
+By default, APIcast applies the `apicast` policy to all the services. This
+policy includes all the functionality offered by APIcast (mapping rules
+matching, authorization and reporting against 3scale backend, etc.). In the
+future, this policy will be split and each of the resulting policies will be
+replaceable with custom ones.
+
+
+## Write your own policy
+
+To write your own policy you need to write a Lua module that instantiates a
+[Policy](../gateway/src/apicast/policy.lua) and defines a method for each of
+the phases where it needs to execute something.
+
+Suppose that we wanted to run a policy that logged some message in the
+`rewrite` and the `header_filter` phases. This is how our module would look
+like:
+```lua
+local policy = require('apicast.policy')
+local _M = policy.new('My custom policy')
+
+function _M:rewrite()
+  ngx.log(ngx.INFO, 'Passing through the rewrite phase.')
+end
+
+function _M:header_filter()
+  ngx.log(ngx.INFO, 'Passing through the header_filter phase')
+end
+
+return _M
+```
+
+If we wanted to read or write in the `context`:
+```lua
+function _M:rewrite(context)
+  -- Read something from the context.
+  local smth = context.something
+
+  -- Write 'b' in the context so other policies or later phases can read it.
+  context.b = 'something_useful_to_be_shared'
+end
+```
+
+Policies can also have a configuration:
+```lua
+local policy = require('apicast.policy')
+local _M = policy.new('My custom policy')
+local new = _M.new
+
+function _M.new(config)
+  local self = new()
+
+  -- 'config' contains two params that define the behaviour of our policy and
+  -- we want to access them from other phases.
+  self.option_a = config.option_a
+  self.option_b = config.option_b
+
+  return self
+end
+```
+
+In the [policy folder](../gateway/src/apicast/policy) you can find several
+policies. These ones are quite simple and can be used as examples to write your
```

Comment:
"They are quite..."

Reply:
I don't think that's correct, because I'm not listing all of them, just some examples.